beep/dataset.py

```diff
@@ -400,39 +408,45 @@ def get_threshold_targets(dataset_diagnostic_properties,
         elif run_target_df['fractional_metric'].min() > threshold and extrapolate_threshold:
             fill_value = "extrapolate"
             bounds_error = False
-            x_throughput_linspace = np.linspace(x_throughput_axis.min(), 2 * x_throughput_axis.max(), num=1000)
-            x_cycle_linspace = np.linspace(x_cycle_axis.min(), 2 * x_cycle_axis.max(), num=1000)
+            x_linspaces = []
+            for axis in x_axes:
+                x_linspaces.append(np.linspace(axis.min(), 2 * axis.max(), num=1000))
```

Comment:
Why does the linspace upper bound go to 2*axis.max() instead of just axis.max()?

Reply:
Ah I see now that this is for cases where the fractional_metric has not yet crossed the threshold. I see why there should be a factor greater than 1, but a value of 2 feels arbitrary.
If a test is very early in its lifetime and extrapolation is on, then 2*axis.max() may still not provide a value that will go below the threshold.
Could it be made more specific such as by making the upper bound the predicted linear extrapolation:
x_thresh_extrap = (threshold - y1)*(x2-x1)/(y2-y1)+x1, where
 y1 = y_interpolation_axis.iloc[-2]
y2 = y_interpolation_axis.iloc[-1]
x1 = axis[-2]
x2 = axis[-1]
We could make the extrapolation go a little beyond what's exactly necessary by subtracting another 0.1 from threshold, for example.